tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js

```diff
@@ -77,6 +104,91 @@ module.exports = {
+
+      let permission = ''
+      this
+        .useXpath()
+        .waitForElementVisible(informationSelector)
+        .click(informationSelector)
+        .waitForElementVisible(selectRoleButton)
+
+      for (let i = 0; i < collaboratorPermissionArray.length; i++) {
+        permission = collaboratorPermissionArray[i]
+        const permissionXpath = this.permissionXpathConverter(permission)
+        await this.api.waitForElementVisible({ selector: permissionXpath, timeout: 100, abortOnFailure: false }, function (result) {
+          if (result.value === true) {
+            this
+              .getAttribute(permissionXpath, 'data-state', (state) => {
+                if ((state.value === 'on' && !requiredPermissionArray.includes(permission)) ||
+                  (state.value === 'off' && requiredPermissionArray.includes(permission))) {
+                  // need to click
+                  this.useXpath()
+                    .waitForElementVisible(permissionXpath)
+                    .click(permissionXpath)
+                }
+              })
+          }
+        })
+      }
+
+      return this.waitForElementVisible('@saveShareButton')
+        .click('@saveShareButton')
+        .waitForOutstandingAjaxCalls()
+        .waitForElementNotPresent('@saveShareButton')
+    },
+    /**
+     *
+     * @param {String} collaborator
+     * @param {String} permissions
+     */
+    assertPermissionIsDisplayed: async function (collaborator, permissions = undefined) {
+      const informationSelector = util.format(this.elements.collaboratorInformationByCollaboratorName.selector, collaborator)
+      const selectRoleButton = informationSelector + this.elements.selectRoleButtonInCollaboratorInformation.selector
+
+      let permissionArray
+      if (permissions !== undefined) {
+        permissionArray = this.permissionArray(permissions)
+      } else {
+        permissionArray = collaboratorPermissionArray
+      }
+
+      this
+        .useXpath()
+        .waitForElementVisible(informationSelector)
+        .click(informationSelector)
+        .waitForElementVisible(selectRoleButton)
+
+      for (let i = 0; i < permissionArray.length; i++) {
+        const permissionXpath = this.permissionXpathConverter(permissionArray[i])
+        if (permissions !== undefined) {
+          await this
+            .assert
+            .attributeEquals(
+              permissionXpath, 'data-state', 'on', `data-state of xpath ${permissionXpath} is not set `)
+        } else {
+          return this.api.isVisible(permissionXpath, result => {
+            if (result.value === true) {
+              this
+                .assert
+                .attributeEquals(permissionXpath, 'data-state', 'off', `data-state of xpath ${permissionXpath} is set `)
+            } else {
+              this
+                .assert
+                .attributeEquals(permissionXpath, 'data-state', 'null', `data-state of xpath ${permissionXpath} found `)
```

Comment:
you do not need this check as the permission is not visible, you have checked that before